src/view/components/hookFormReactSpectrum/picker.jsx

```diff
@@ -29,7 +29,9 @@ const Picker = ({ name, description, width, ...otherProps }) => {
         {...otherProps}
         selectedKey={value}
         onSelectionChange={setValue}
-        onBlur={onBlur}
```

Comment:
The `onBlur` from Formik expects a native blur event to be passed. Since React-Spectrum doesn't provide the native event, the field wasn't being marked as touched. `setTouched` is designed more for custom components like we have here.